api/services/registration_service.py

```diff
@@ -88,18 +88,21 @@ def update_registration(self, registration, jsonData):
         if not registration:
             return
 
-        registration.program_id = self.selected_program_id
+        if not (registration.program_id and self.has_default_program_selection):
+            registration.program_id = self.selected_program_id
 
```

Comment:
Update registration program id only if user selected any option or if the program id is null in the database.